src/components/RenderRoles/RenderRoles.js

```diff
@@ -89,7 +91,7 @@ class RenderRoles extends React.Component {
       >
         <IfConsortium>
           <IfConsortiumPermission perm="consortia.user-tenants.collection.get">
-            {Boolean(affiliations?.length) && (
+            {Boolean(affiliations?.length) && isAffiliationsVisible && (
```

Comment:
> Workaround: refresh the page

Why does refreshing the page solve this issue?

Reply:
Because after switching the user's type, an array of affiliations starts caching. That's why I added an additional condition as it's done in `Affiliation accordion`
https://github.com/folio-org/ui-users/blob/master/src/views/UserDetail/UserDetail.js#L763